waku/v2/node/storage/message/sqlite_store/sqlite_store.nim

```diff
@@ -19,6 +19,10 @@ logScope:
 
 
 proc init(db: SqliteDatabase): MessageStoreResult[void] =
+  ## Misconfiguration can lead to nil DB
+  if db.isNil():
+    return err("db not initialized")
+
```

Comment:
Mmm. I think if we ensure that a SQLite store cannot be initialized with a nil DB, the isNil() check on the rest of its API (incl. but not only for `close()` becomes redundant?

Reply:
You are totally right. At the begining I was thinking that the following could be (rare but) possible.

```nim
let nilStore = SqliteStore(db: nil, insertStmt: nil)
nilStore.close() # <-error!
```

However just learned that if `db` and `insertStmt` are not exported with `*`, you cant create a new `SqliteStore` with `SqliteStore()` (`the field 'xxx' is not accessible`). So you need to use the "constructor" `.init()`, that ensures `db` won't be `nil` when closing it.

I like defensive programming, but perhaps its too much here since we are alread protected by the `init`.